### PR TITLE
fix: skip cellbase short-id collision validation

### PR DESCRIPTION
**Pull request by quake**

### CODE REVIEW

File: sync/src/relayer/compact_block_verifier.rs [hunk 1]
```diff
@@ -75,9 +75,11 @@ impl ShortIdsVerifier {
             return Err(Error::Misbehavior(Misbehavior::DuplicatedShortIds));
         }
 
-        // Check intersection of prefilled transactions and short ids
+        // Check intersection of prefilled transactions and short ids.
+        // Cellbase is skipped since it's always prefilled and has the chances of collision with other txs
         let is_intersect = prefilled_transactions
             .into_iter()
+            .skip(1)
```
Comment (u2) [inline sync/src/relayer/compact_block_verifier.rs]: The `cellbase` is always in the first position? I am not sure about that.

Comment (quake) [reply sync/src/relayer/compact_block_verifier.rs]: Yes
https://github.com/nervosnetwork/ckb/blob/a875ce11246df4acfdd047b5bc114eede2ee64ed/sync/src/relayer/compact_block_verifier.rs#L26-L33

Comment (u2) [reply sync/src/relayer/compact_block_verifier.rs]: ok

**Comment by quake** [comment]
bors retry

**Pull request merged**